portfolio/src/main/java/com/google/sps/servlets/InteractiveChartServlet.java

```diff
@@ -10,27 +16,46 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** TODO(briafassler): store data to a database */
+/** Servlet for the interactive pie chart */
 @WebServlet("/interactive-chart")
 public class InteractiveChartServlet extends HttpServlet {
   private Map<String, Integer> emotionVotes = new HashMap<>();
-  private static String EMOTION_DATA_PARAM = "emotion";
+  private static String EMOTION_DATA = "emotion";
+  private static String VOTES_PARAM = "Votes";
+  private static String TIMESTAMP_PARAM = "timestamp";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Gson gson = new Gson();
-    String json = gson.toJson(emotionVotes);
+    Query votesQuery = new Query(VOTES_PARAM).addSort(TIMESTAMP_PARAM, SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(votesQuery);
+
+    String votesJSON = new Gson().toJson(emotionVotes);
     response.setContentType("application/json");
-    response.getWriter().println(json);
+    response.getWriter().println(votesJSON);
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String emotion = request.getParameter(EMOTION_DATA_PARAM);
+    String emotion = request.getParameter(EMOTION_DATA);
     int currentVotes = emotionVotes.containsKey(emotion) ? emotionVotes.get(emotion) : 0;
     emotionVotes.put(emotion, currentVotes + 1);
+    long timestamp = System.currentTimeMillis();
+
+    // Stores object in datastore entity
+    Entity votesEntity = new Entity(VOTES_PARAM);
```

Comment:
you should name the constant as TABLE_NAME instead to not get the constant confused with anything else

Reply:
done